### PR TITLE
Fix build against cuda<=11

### PR DESCRIPTION
**Pull request by bartoszek**
Have some trouble building against `cuda:11` so i reworked nvcc compute arch section a bit, now it looks clearer and nicer.
_btw. is there a way to force build for specific compute capability, can't find it anywhere?_

### CODE REVIEW

**Comment by simogasp** [comment]
@griwodz maybe we can take the cmake function you did for popsift here as well?

**Comment by griwodz** [comment]
> Have some trouble building against `cuda:11` so i reworked nvcc compute arch section a bit, now it looks clearer and nicer.
> _btw. is there a way to force build for specific compute capability, can't find it anywhere?_

To use only a single CC (and save a lot of compile time), I'm usually running the usual cmake first and I edit the ALICEVISION_CUDA_CC_LIST variable in the CMakeCache.txt afterwards (I use ccmake, but manual should also work).
You can probably also add -DALICEVISION_CUDA_CC_LIST=<your cc> to the first cmake call for the same result, but I'm not sure.

**Comment by griwodz** [comment]
> @griwodz maybe we can take the cmake function you did for popsift here as well?

@simogasp  Should I try to do it in this pull request? Or take the current PR as an urgent fix because more people are upgrading to SDK 11 and create a new PR to add the ChooseCudaCC.cmake file?

**Comment by simogasp** [comment]
I think you can go directly with this one, we fix it properly once for all

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -730,9 +730,13 @@ if(ALICEVISION_HAVE_CUDA)
 
   set(CUDA_NVCC_FLAGS_DEBUG   "${CUDA_NVCC_FLAGS_DEBUG};-G;-g")
   # set(CUDA_NVCC_FLAGS_RELEASE "${CUDA_NVCC_FLAGS_RELEASE};-O3")
-  if(NOT CUDA_VERSION_MAJOR LESS 9)
-    set(ALICEVISION_CUDA_CC_LIST_BASIC 30 35 50 52 60 61 62)
-  elseif(CUDA_VERSION_MAJOR GREATER 7)
+  if(CUDA_VERSION_MAJOR GREATER_EQUAL 11)
+    set(ALICEVISION_CUDA_CC_LIST_BASIC 50 52 60 61 62 70 72 75 80)
```
Comment (griwodz) [inline src/CMakeLists.txt:734]: CC 50 is also on the deprecated list for SDK 11 (according to the 11.0 release notes and the warnings I'm getting).  We should either keep CC 35 as well, or drop both CC 35 and CC 50. I'm in favour of keeping 35.

**Comment by bartoszek** [comment]
> You can probably also add -DALICEVISION_CUDA_CC_LIST=YourCC
Thanks @griwodz , somehow I thought it would be overwritten 🤦

**Comment by griwodz** [comment]
> > You can probably also add -DALICEVISION_CUDA_CC_LIST=YourCC
> > Thanks @griwodz , somehow I thought it would be overwritten 🤦

@bartoszek The set() operation is in an if() and there's no CACHE label, so you're probably right and it is overwritten. It will work with the cmake file that we'll adapt from PopSift in a followup-PR, but we should merge your PR first because it fixed SDK 11 now and not next week. ;-)

**Comment by bartoszek** [comment]
> you're probably right and it is overwritten

I've tested it with my [AUR package and it works fine, I can build for `compute_capabilities:50,52` with `makepkg CUDA_ARCH="50 52"` so it isn't overwritten](https://aur.archlinux.org/cgit/aur.git/commit/?h=alice-vision-git&id=2401adac6a9e856f9f6c56681152cd4910242a39)

**Comment by bartoszek** [comment]
Sorry, accidentally push some local stuff here - already rolled back.

**Comment by simogasp** [comment]
thx for the contribution!

**Pull request merged**